CONTRIBUTING.md

```diff
@@ -1,65 +1,251 @@
-   and discuss Flutter][discord]
-
-Happy Fluttering!
-
-[issues]: https://github.com/flutter/flutter/issues
-[PRs]: https://github.com/flutter/flutter/pulls
-[discord]: https://discord.gg/rflutterdev
-[doc-issues]: https://github.com/flutter/website/issues
-[doc-PRs]: https://github.com/flutter/website/pulls
-[so]: https://stackoverflow.com/tags/flutter
-[mailinglist]: https://groups.google.com/d/forum/flutter-dev
-[chat]: https://github.com/flutter/flutter/wiki/Chat
-[reddit]: https://www.reddit.com/r/FlutterDev
+If you would like to contribute to the Flutter website,
+we’re happy to have your help! Anyone can contribute,
+regardless of your writing experience or how you view yourself as a writer,
+developer, or grammar stickler.
+
+## Contributors
+
+Before submitting a PR,
+[search for an issue][search-issue] that covers your concern.
+If one exists, add `Fixes #<found-issue>` to your new PR.
+If you don't find one and your change is significant,
+consider [creating an issue][create-issue].
+The issue should describe the problem, how you would plan to fix it,
+and should be assigned to yourself.
+
+When you create a new page or perform a major rewrite of an existing page,
+consider drafting it first in Google Docs to simplify collaboration.
+Once your Google doc has been created,
+add a link to it from your new GitHub issue.
+
+### Draft changes
+
+1. [Create a branch][create-branch] for your change.
+   Consider naming it `fix-<issue_number>` to help others find the related issue.
+   The new branch should be based on the `main` branch or a feature
+   or version branch.
+
+   To branch and set upstream at the same time, use this command:
+
+   ```terminal
+   git checkout -b <fix-issue> upstream main
+   ```
+
+1. Try to limit content changes to those that address the request made in the
+   associated issue or doc plan. Use your best judgment.
+
+   1. Avoid adjusting indents, line breaks, formatting, and the like
+      unrelated to your PR as part of the PR whenever possible.
+      To separate work unrelated to the PR, use a new commit,
+      a [separate issue][create-issue], a [separate PR](#submit-pr),
+      or some combination.
+
+   1. Explain the reason for the additional changes in the PR summary,
+      if necessary. For guidance on what to include in a PR summary,
+      check out [this section in the Blockly docs][blockly-pr-note]
+      or this [PullRequest.com blog post][good-pr-blog-post].
+
+1. Make sure all copy and content meets the standards in the
+   [Google Developer Documentation Style Guide][style-guide].
+   Focus on the [**Voice and tone**][gs-tone]
+   and the [**Highlights**][gs-highlights]
+   pages. They provide the best starting points.
+
+1. Using `git`, [commit changes](https://git-scm.com/docs/git-commit) often.
+   It simplifies reverting a specific change.
+   Reviewers can choose to review changes made since their last review and
+   shorten review cycles.
```

Comment:
I'm not sure what this means. Aren't contributors required to use git? What's the alternative?